javascript/organizing_your_javascript_code/factory_functions_and_module_pattern.md

```diff
@@ -1,305 +1,250 @@
 
-A quick note about line 3 from the factory function example. In 2015, a handy new shorthand for creating objects was added into JavaScript. Without the shorthand, line 3 would have looked something like this:
+The best way to approach this would be to start with an example - take a look at this piece of code below.
 
 ~~~javascript
-return {name: name, age: age, sayHello: sayHello};
+function makeAdding (first) { // "first" is scoped within the makeAdding function
+  return function resulting (second) { // "second" is scoped within the resulting function
+    return first + second;
+  }
+} // but we've not seen an example of a **function** being returned, thus far - how do we use it?
+
+const add5 = makeAdding(5);
+console.log(add5(2)) // logs 7
 ~~~
 
-Put simply, if you are creating an object where you are referring to a variable that has the exact same name as the object property you're creating, you can condense it like so:
+A lot going on, so let's break it down:
 
-~~~javascript
-return {name, age, sayHello};
-~~~
+1. The `makeAdding` function takes an argument, `first`, and returns another **function** (which we have named as `add5` while using).
+1. When an argument is passed to the returned function (in our case, `add5`), it returns the result of adding up the number passed earlier to the number passed now (`first` to `second`).
 
-With that knowledge in your pocket, check out this little hack:
+Now, while it may sound good at first glance, you can already be raising your eyebrows at the second statement. As we've learned, the `first` variable is scoped within the `makeAdding` function. When we declare and use `add5`, however, we're **outside** the `makeAdding` function. How does the `first` variable still exist, ready to be added when we pass an argument to the `add5` function? This is where we encounter the concept of closures.
 
-~~~javascript
-const name = "Maynard";
-const color = "red";
-const number = 34;
-const food = "rice";
-
-// logging all of these variables might be a useful thing to do,
-// but doing it like this can be somewhat confusing.
-console.log(name, color, number, food); // Maynard red 34 rice
-
-// if you simply turn them into an object with brackets,
-// the output is much easier to decipher:
-console.log({name, color, number, food});
- // { name: 'Maynard', color: 'red', number: 34, food: 'rice' }
-~~~
+Functions in Javascript form closures. A closure refers to the combination of a function and the **surrounding state** within which a function was declared in. This surrounding state, also called its lexical environment, consists of any local variables that were in scope at the time the closure was made. Here, `add5` is a reference to the `resulting` function, created when the `makeAdding` function is executed, thus it has access to the lexical environment of the `resulting` function, which contains the `first` variable, making it available for use, rather than deleting it for being out of scope.
```

Comment:
small typo and I also think `enclosing context` is a better term here
```suggestion
Functions in JavaScript form closures. A closure refers to the combination of a function and the **enclosing context** within which a function was declared in. This enclosing context, also called its lexical environment, consists of any local variables that were in scope at the time the closure was made. Here, `add5` is a reference to the `resulting` function, created when the `makeAdding` function is executed. Therefore, `add5` has access to the lexical environment of the `resulting` function, which contains the `first` variable, making it available for use, rather than deleting it for being out of scope.
```

I also think we can try to elaborate more on this:
> rather than deleting it for being out of scope
"deleting" what? What mechanism does it? What "deletes" it. It's something we can try making a small note of as this kind of behavior is crucial in React as well.

Reply:
Can you point out what to write instead? Cause I am not sure about how elaborating can be done while keeping it simple to understand. @fcasibu

Reply:
@Lofty-Brambles I also can't think of a simple explanation to expand on this 😓 but here's what I can think of to rephrase:

```suggestion
Functions in JavaScript form closures. A closure refers to the combination of a function and the **surrounding state** within which a function was declared in. This surrounding state, also called its lexical environment, consists of any local variables that were in scope at the time the closure was made. Here, `add5` is a reference to the `resulting` function, created when the `makeAdding` function is executed. Therefore, instead of being removed from memory due to going out of scope, the `first` variable remains accessible within the lexical environment of the `resulting` function, allowing it to be used by `add5`.
```
What do you think?

Reply:
LGTM!